integr8ly/grafana-operator
# Add configurable service names and add missing MaxLines for Loki DS

## Description

- Adds missing `MaxLines` for Loki data sources, as per community request.
- Adds missing `DerivedFields` for data sources, as per community request.
- Adds a configurable service name for grafana services:
  - If `spec.service` is not empty and the `name` field is not an empty string, the operator will use this as the default name for the service relating to this specific grafana, otherwise a default `Grafana-Service` name is used.

## Relevant issues/tickets

#306 
#310 
#324 


## Type of change




## Checklist


## Verification steps


- Checkout the master branch and run `make cluster/prepare/local` .
- Run the Operator (either deploy or run locally)
- Ensure that the operator creates the grafana service correctly using the default `Grafana-Service` name.
- Stop the operator.
- Checkout this branch locally.
-  Edit Grafana.yaml and add the following like to the bottom of the `Spec` field
```yaml
  service:
    name: "newServiceName"
```
- Reapply the CR to the cluster
- Run the operator
- Ensure the new service is created and the name is propagated and that no other services exist, i.e with a previous name
